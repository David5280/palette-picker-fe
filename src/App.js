import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Palette! Picker</h1>
      </header>
      <main>
        <section>
          <div>Palette 1</div>
          <div>Palette 2</div>
          <div>Palette 3</div>
          <div>Palette 4</div>
          <div>Palette 5</div>
          <button>Generate new Palette!</button>
          <form>
            <select>
              <option>Palette 1</option>
              <option>Palette 2</option>
              <option>Palette 3</option>
            </select>
            <input type='text' placeholder='Palette Name' />
            <button>Save Palette</button>
          </form>
        </section>
        <section>
          <form>
            <label for='new-project-name'>Create a new project:</label>
            <div>
              <input type='text' placeholder='New Project Name...' />
              <button>Save Project</button>
            </div>
          </form>
        </section>
        <section>
          <p>Project 1</p>
          <p>Project 2</p>
        </section>
      </main>
    </div>
  );
}

export default App;
