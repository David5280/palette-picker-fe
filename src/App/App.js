import React from 'react';
import Palette from '../Palette/Palette';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import { serverCall } from '../fetchCalls/fetchCalls';
import { connect } from 'react-redux';
import { retrieveProjects, loadComplete, hasErrored } from '../actions';

class App extends React.Component {
  constructor() {
    super()
    this.state = {}
  }

  componentDidMount = async () => {
    try {
      const projects = await serverCall('projects')
      await this.props.retrieveProjects(projects)
      await this.props.loadComplete()
    } catch (error) {
      this.props.loadComplete()
      this.props.hasErrored(error.message)
    }
  }

  setColors = (colors) => {
    this.setState({colors})
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1>Palette! Picker</h1>
        </header>
        <main>
          <Palette />
          <ProjectContainer />
        </main>
        <footer>
          <p>Follow us on the interwebs, nerd.</p>
        </footer>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  retrieveProjects: projects => dispatch(retrieveProjects(projects)),
  loadComplete: () => dispatch(loadComplete()),
  hasErrored: error => dispatch(hasErrored(error))
});

export default connect(null, mapDispatchToProps)(App);
