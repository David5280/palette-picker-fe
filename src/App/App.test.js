import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'

describe('Project Container', () => {
  
  it('renders without crashing', () => {
    let wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should do something', () => {

  })
})

