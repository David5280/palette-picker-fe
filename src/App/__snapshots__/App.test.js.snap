// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Container renders without crashing 1`] = `ShallowWrapper {}`;

exports[`renders without crashing 1`] = `ShallowWrapper {}`;
